### ISSUE TITLE
Deploy nuget.config with encrypted ApiKey value via Chef

### ISSUE THREAD

**Issue by pvandervelde** [NuGet/Home#1846]
I'm trying to deploy a nuget.config file to all of our build machines using powershell and Chef. Copying the standard nuget.config file across can be done without issues. The build agents also need to be able to push packages up to our company repository for which they need an API key. Adding the encrypted API key to the nuget.config file can be done by creating the configuration file once and then copying that, however when I store the API key in a nuget.config file on two different machines I get two different values in the configuration file. This leads me to believe that machine specific information is used in the encryption.
So now the question is how do I achieve my goal of deploying the nuget.config files to the build machines while avoiding having to:
a) manually add the apikey to the config file on each machine
b) run scripts or code as the build server user instead of the machine admin

In other words I want the configuration of my build agent to be completed automated but I can't figure out how to handle the API key in the nuget.config file.

**Comment by yishaigalatzer** [comment]
The encryption is done by user key. So you will have to either pass the API key to the push command using the -ApiKey flag, and roll your own encryption/access to the ApiKey in the build script from a central source. OR run nuget config SetApiKey as the user (which you don't seem to want to do).
